src/common/transformations/tests/op_conversions/sdpa_to_paged_attention_test.cpp

```diff
@@ -0,0 +1,537 @@
+        auto rope_emb_cos = Qwen7bChatPA::gen_rope_emb_cos(max_context_len_aligned, position_ids_aligned);
+
+        // rope Q, K:
+        auto rope_Q = Qwen7bChatPA::gen_rope(QKV::Q, qkv_proj, head_size, rope_emb_sin, rope_emb_cos);
+        auto rope_K = Qwen7bChatPA::gen_rope(QKV::K, qkv_proj, head_size, rope_emb_sin, rope_emb_cos);
+
+        // Current/past/total Seq lengths calculation:
+        auto current_seq_len = Qwen7bChatPA::gen_current_len(rope_K);
+        auto past_seq_len = Qwen7bChatPA::gen_past_len(input_ids_aligned, max_context_len);
+        auto total_seq_len = Qwen7bChatPA::gen_total_len(current_seq_len, past_seq_len);
+
+        // Q, K, V:
+
+        shared_ptr<Node> head_size_2;
+        auto Q = Qwen7bChatPA::gen_Q(total_seq_len, rope_Q);
+        auto K = Qwen7bChatPA::gen_K(rope_K);
+        auto V = Qwen7bChatPA::gen_V(qkv_proj, head_size_2);
+
+        // Additional PA arguments:
+        auto sliding_window = std::make_shared<v0::Constant>(element::i32, Shape{}, 0);
+        auto alibi_slopes = std::make_shared<v0::Constant>(element::f32, Shape{0});
+        auto scale = std::make_shared<v0::Constant>(element::f32, Shape{}, MOCK_VALUE);
+
+        // PagedAttention:
+        auto pa = std::make_shared<op::PagedAttentionExtension>(OutputVector{Q,
+                                                                             K,
+                                                                             V,
+                                                                             key_cache_0,
+                                                                             value_cache_0,
+                                                                             past_lens,
+                                                                             subsequence_begins,
+                                                                             block_indices,
+                                                                             block_indices_begins,
+                                                                             scale,
+                                                                             sliding_window,
+                                                                             alibi_slopes,
+                                                                             max_context_len});
+        pa->set_out_type(0, element::i64);
+        auto pa_aligned = Qwen7bChatPA::align_pa_layout(pa, head_size_2);
+        auto res = makeOP<v0::Result>({pa_aligned});
+
+        model_ref = std::make_shared<ov::Model>(OutputVector{res}, params);
+    }
+    // TODO: align precisions, check the copying of "fuse_names" attr in SDPAToPagedAttention
+    // checking the graph structure and names, other checks are temporarily disabled:
+    comparator.disable(FunctionsComparator::PRECISIONS);
+    disable_rt_info_check();
+}
```

Comment:
Let me guess: coding OV models as a C++ code in tests is the main reason why the tests compilation is so time-consuming, right? It is a joke (I hope), but why do we need to code all these parts here as C++ code but not just `read_model` them from IR files located somewhere? I'm not against representing parts of the models in the code, I'm just concerned that you are spending more time preparing this code in comparison to the code that reads from an IR file. You are using a tool to generate a body for each part, but you still need to organize parts as a whole. Or are you also using a tool to do that as well?

Reply:
I used PrintModel transformation to generate the c++ code of the model.
But I had to split it by parts (attention mask, Q, K, V etc) manually. 
Not so sure if we have a way to automate it. Unfortunately, even the famous ov2py tool can't do it.
I'm planning to write the tests for the matcher passes as well so I believe the splitting was not so useless.